inc/user_templates/parameters/commonitilobjectparameters.class.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2021 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\User_Templates\Parameters;
+
+use CommonDBTM;
+use CommonITILActor;
+use CommonITILObject;
+use Entity;
+use Glpi\User_Templates\Parameters\Parameters_Types\ArrayParameter;
+use Glpi\User_Templates\Parameters\Parameters_Types\AttributeParameter;
+use Glpi\User_Templates\Parameters\Parameters_Types\ObjectParameter;
+use Group;
+use ITILCategory;
+use User;
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+/**
+ * Parameters for "CommonITILObject" items
+ */
+class CommonITILObjectParameters extends AbstractTemplatesParameters
+{
+   public static function getTargetClasses(): array {
+      return [CommonITILObject::class];
+   }
+
+   public function defineParameters(): array {
+      return [
+         new AttributeParameter("id", __("Ticket's id")),
```

Comment:
Should probably add a type parameter so the user can define logic based on the ITILObject type (Ticket, Change, Problem).

Reply:
Done 8bf9d05b46.